### ISSUE TITLE
NoMethodError: [!] undefined method `real_path' for nil:NilClass

### ISSUE THREAD

**Issue by StevenMasini** [CocoaPods/Xcodeproj#625]
Hi xcodeproj contributors,

Some days ago my `fastlane` script failed. So I initially opened an [issue](https://github.com/fastlane/fastlane/issues/13667) on the `fastlane` repository.

But it ended up being an issue related to `xcodeproj`. So I am forwarding the issue here.

[NoMethodError: [!] undefined method `real_path' for nil:NilClass #13667](https://github.com/fastlane/fastlane/issues/13667)

TL;DR I had a dead reference to a library into my `project.xcodeproj`.

I think the best way to understand is still to read the whole thread.
The good news is that we finally figured out why `xcodeproj` crashes, and it crashes for a good reason.

But maybe we could make the reason of this crash more obvious in case it happens to others ?

**Comment by djmixroman** [comment]
Same with "empty?" method
